runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/clues/ThreeStepCrypticClue.java

```diff
@@ -147,6 +185,36 @@ private boolean checkForPart(final ItemContainer itemContainer, int clueScrollPa
 		return false;
 	}
 
+	private void propagateRequirements(final Map.Entry<CrypticClue, Boolean> clueStep)
```

Comment:
Personally I prefer not to pass around Map.Entry values since they effectively become unnamed tuples and are harder to read than just unpacking the value into named parameters